File: Counter/src/main/java/com/kchmielewski/spring/counter/logic/VisitCounterService.java
package com.kchmielewski.spring.counter.logic;

import com.kchmielewski.spring.counter.model.ApplicationCounter;
import com.kchmielewski.spring.counter.model.Counter;
import com.kchmielewski.spring.counter.model.SessionCounter;
import com.kchmielewski.spring.counter.web.VisitCounterResponse;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Service;

@Service
public class VisitCounterService {
    private final ApplicationContext applicationContext;
    private final Counter applicationCounter;

    public VisitCounterService(ApplicationContext applicationContext, ApplicationCounter counter) {
        this.applicationContext = applicationContext;
        applicationCounter = counter;
    }

    public VisitCounterResponse count() {
        return new VisitCounterResponse(applicationCounter.count(),
                applicationContext.getBean(SessionCounter.class).count());
    }

    public VisitCounterResponse visit() {
        return new VisitCounterResponse(applicationCounter.visit(),
                applicationContext.getBean(SessionCounter.class).visit());
    }
}

File: Counter/src/main/kotlin/com/kchmielewski/kotlin/spring/counter/logic/VisitCounterService.kt
package com.kchmielewski.kotlin.spring.counter.logic

import com.kchmielewski.kotlin.spring.counter.model.ApplicationCounter
import com.kchmielewski.kotlin.spring.counter.model.SessionCounter
import com.kchmielewski.kotlin.spring.counter.model.Visits
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Service

@Service
class VisitCounterService(private val applicationContext: ApplicationContext,
                          private val applicationCounter: ApplicationCounter) {

    fun count(): Visits = Visits(applicationCounter.count(),
            applicationContext.getBean(SessionCounter::class.java).count())

    fun visit(): Visits = Visits(applicationCounter.visit(),
            applicationContext.getBean(SessionCounter::class.java).visit())
}
